### PR TITLE
[feat] Import SRT/WebVTT subtitle files as caption clips

### PR DESCRIPTION
**Pull request by htin1**
Screenshot 2026-08-08 at 11 10 21 PM

## Summary

Adds SRT and WebVTT subtitle support:

1. importing a `.srt`/`.vtt` file creates a **subtitle media asset** in the media panel
2. selecting it renders the parsed cue list in the preview area
3. dragging it onto the timeline places its cues as **caption text clips synced to the file's authored timecodes**
4. The agent can do the same via `import_media` and then `add_captions` with `subtitleMediaRef`.

## Approach

- **Parser** (`SubtitleFileParser`): tolerant of real-world files — BOM, CRLF, dot milliseconds, missing SRT indices, out-of-order cues, WebVTT cue settings and NOTE/STYLE/REGION blocks, hourless timestamps. Inline markup (`<i>`, `<v>`, `{\an8}`, inline timestamps) is stripped and entities decoded; UTF-8 with Latin-1 fallback. Malformed timing fails with the offending line number.
- **Asset-only type**: `ClipType.subtitle` routes the normal import pipeline (open panel, Finder drops, folder scans, relink, `import_media`), but is never a clip's `mediaType` — the drop plan skips subtitle assets, and `add_clips`/`insert_clips` refuse them with pointers to the caption path. Duration = last cue end, loaded during finalize, so malformed files surface as unprocessable at import.
- **Single placement path**: drag, Finder drop, and the agent tool all funnel into `importCaptions(from:)` → `CaptionSpecBuilder.build(cues:…)` → `placeCaptionTrack`, reusing the caption transform math and overlap resolution (gap-closing disabled to honor file timing). One caption group, one "Add Captions" undo step per file; restyleable with existing caption tooling (`update_text`, caption group summaries).
- **Rejected alternatives**: direct-to-timeline import (FCP-style, no asset — first iteration, replaced at user request); drop-position placement (Resolve manual-drag style — implemented, then removed in favor of timecode sync for simplicity).

```mermaid
flowchart LR
    srtFile[".srt / .vtt"] -->|"import (UI or import_media)"| asset["Subtitle MediaAsset"]
    asset -->|select| preview["Cue-list preview"]
    asset -->|"drag / add_captions subtitleMediaRef"| captions["Caption clips at authored timecodes, one undo step"]
```

## Testing

- `swift build` — clean.
- `swift test --filter "Subtitle|AddCaptions|Caption"` — 78 tests, 19 suites, all passing: parser edge cases (tolerances, markup, five malformed variants with line numbers), cue→frame conversion, overlap trimming, sub-frame clamping, asset finalize, drop placement + undo round-trip, drop-plan guard, and the agent tool (placement receipt, short-id expansion, combined-option/wrong-type/malformed refusals).
- `scripts/localization/sync.sh` — clean.
- MCP end-to-end against the running app: `import_media` for 7 well-formed + 4 malformed fixtures (Latin-1, BOM/CRLF, markup, VTT features, sub-frame, overlaps), `inspect_media` cue verification, `add_captions subtitleMediaRef` placement, independent `get_timeline` read-back, single-step `undo`. The live run caught and fixed a short-id expansion gap.
- Manual UI: drag-to-timeline verified by @harrison. Remaining manual checks before ready-for-review: cue-list preview rendering (well-formed + malformed), Subtitle filter chip, media-panel tile.

## Change statistics

- Code: ~490 insertions (parser 134, preview view 80, placement + asset type + switch arms + agent tool ~275), 103 deletions (mostly `PreviewContainerView` restructure).
- Tests: ~255 insertions across 3 files (parser/spec, import/placement integration, agent tool).
- Generated: localization strings (+7).

---

> [!NOTE]
> **Medium Risk**
> Touches timeline caption placement, import MIME handling, and agent tools across many switch sites; malformed files are surfaced at parse/finalize, but timeline mutation and undo grouping need careful review.
> 
> **Overview**
> Adds **SRT/WebVTT** support end-to-end: new **`ClipType.subtitle`** media assets (import panel, Finder drops, **`import_media`** with `application/x-subrip` / `text/vtt`), a **cue-list preview** when selected, and placement of cues as **caption text clips at the file’s timecodes** (not as timeline media clips).
> 
> **`SubtitleFileParser`** parses real-world SRT/VTT; **`CaptionSpecBuilder.build(cues:)`** maps cues to frames with overlap trimming but **no gap-closing** so authored timing is preserved. UI drops and media import coalesce caption placement into the same undo path as **`importCaptions(from:)`**; subtitle assets are excluded from clip drop plans and blocked on **`add_clips`**.
> 
> The agent gains **`add_captions` `subtitleMediaRef`** (mutually exclusive with transcription options), **`inspect_media`** cue metadata, and updated tool docs/errors. Generation/edit paths explicitly reject subtitle assets.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 474ecd99e7e6cbd8481fb89056c4e8bd98f39615. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: Sources/PalmierPro/Agent/Tools/ToolExecutor+Import.swift [hunk 1]
```diff
@@ -82,7 +82,7 @@ extension ToolExecutor {
         }
         let ext = fileURL.pathExtension.lowercased()
         guard let type = ClipType(fileExtension: ext) else {
-            throw ToolError("Unsupported file extension '.\(ext)'. Supported: mov/mp4/m4v, mp3/wav/aac/m4a/aiff/aifc/caf/flac, png/jpg/jpeg/tiff/heic, json (Lottie).")
+            throw ToolError("Unsupported file extension '.\(ext)'. Supported: mov/mp4/m4v, mp3/wav/aac/m4a/aiff/aifc/caf/flac, png/jpg/jpeg/tiff/heic, json (Lottie), srt/vtt (subtitles).")
```
[the remark this answers is not in the document]
Comment (htin1) [reply Sources/PalmierPro/Agent/Tools/ToolExecutor+Import.swift:85]: Fixed in 474ecd99: application/x-subrip, text/srt, and text/vtt now map to subtitle assets for bytes and MIME-overridden URL imports, with a bytes round-trip test.

File: Sources/PalmierPro/Editor/ViewModel/EditorViewModel+MediaLibrary.swift [hunk 2]
```diff
@@ -429,6 +429,7 @@ extension EditorViewModel {
         } else {
             addClipsWithSettingsCheck(assets: placeable, operation: operation)
         }
+        await placeCaptions(fromSubtitleAssets: placeable.filter { $0.type == .subtitle })
```
[the remark this answers is not in the document]
Comment (htin1) [reply Sources/PalmierPro/Editor/ViewModel/EditorViewModel+MediaLibrary.swift]: Fixed in 8815d1be: caption placement now runs inside the drop's deferred operation (single "Add Media" undo group), so it executes after the settings dialog resolves rather than before.

File: Sources/PalmierPro/MediaPanel/CaptionsTab/SubtitleFileParser.swift [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                throw ParseError.missingWebVTTHeader
+            }
+            blocks.removeFirst()
+        }
+
+        var cues: [SubtitleCue] = []
+        for block in blocks {
+            if format == .webVTT, ["NOTE", "STYLE", "REGION"].contains(where: block.lines[0].hasPrefix) {
+                continue
+            }
```
[the remark this answers is not in the document]
Comment (htin1) [reply Sources/PalmierPro/MediaPanel/CaptionsTab/SubtitleFileParser.swift:85]: Fixed in 8815d1be: WEBVTT/NOTE/STYLE/REGION now match on token boundaries, so identifiers like NOTE123 parse as cues; covered by treatsKeywordPrefixedIdentifiersAsCuesNotComments.
